src/odbc_connection.cpp

```diff
@@ -2664,14 +2706,118 @@ bind_buffers
     IGNORED_PARAMETER
   );
 
-  return_code =
-  SQLSetStmtAttr
-  (
-    data->hstmt,
-    SQL_ATTR_ROWS_FETCHED_PTR,
-    (SQLPOINTER) &data->rows_fetched,
-    IGNORED_PARAMETER
-  );
+  // Some drivers don't feel the need to implement the SQL_ATTR_ROW_BIND_TYPE
+  // option for SQLSetStmtAttr, so this call will return an error. Instead of
+  // returning an error, just set "simple_binding" to true, and bind one row
+  // at a time for fetching.
+  if (!SQL_SUCCEEDED(return_code))
+  {
+    if (data->fetch_size == 1)
+    {
+      SQLRETURN   diagnostic_return_code;
+      SQLSMALLINT textLength;
+      SQLTCHAR    errorSQLState[SQL_SQLSTATE_SIZE + 1];
+      SQLINTEGER  nativeError;
+      SQLTCHAR    errorMessage[ERROR_MESSAGE_BUFFER_BYTES];
+
+      diagnostic_return_code =
+      SQLGetDiagRec
+      (
+        SQL_HANDLE_STMT,            // HandleType
+        data->hstmt,                // Handle
+        1,                          // RecNumber
+        errorSQLState,              // SQLState
+        &nativeError,               // NativeErrorPtr
+        errorMessage,               // MessageText
+        ERROR_MESSAGE_BUFFER_CHARS, // BufferLength
+        &textLength                 // TextLengthPtr
+      );
+
+      if (SQL_SUCCEEDED(diagnostic_return_code)) 
+      {
+        if (strcmp("HY092", (const char*)errorSQLState) == 0)
+        {
+          data->simple_binding = true;
+          data->rows_fetched   = 1;
+          return_code          = SQL_SUCCESS;
+        }
+        else
+        {
+          return return_code;
+        }
+      }
+      else
+      {
+        return return_code;
+      }
+    }
+    else
+    {
+      return return_code;
+    }
```

Comment:
Quite a lot of nesting. If you did early returns you could remove most/all of the nestings.